include/swift/ABI/Metadata.h

```diff
@@ -3316,10 +3316,43 @@ struct TargetInPlaceValueMetadataCache {
 using InPlaceValueMetadataCache =
   TargetInPlaceValueMetadataCache<InProcess>;
 
+template <typename Runtime>
+struct TargetResilientClassMetadataPattern;
+
 /// An instantiation pattern for non-generic resilient class metadata.
 /// Used in conjunction with InPlaceValueMetadataInitialization.
 using MetadataRelocator =
-  Metadata *(const TargetTypeContextDescriptor<InProcess> *description);
+  Metadata *(const TargetTypeContextDescriptor<InProcess> *type,
+             const TargetResilientClassMetadataPattern<InProcess> *pattern);
+
+/// An instantiation pattern for non-generic resilient class metadata.
```

Comment:
Maybe this should elaborate a bit about the breakdown between the different class-metadata allocation situations.